Doc/c-api/object.rst

```diff
@@ -197,6 +197,13 @@ Object Protocol
    in favour of using :c:func:`PyObject_DelAttr`, but there are currently no
    plans to remove it.
 
+   The function must not be called with ``NULL`` *v* and an an exception set.
+   This case can arise from forgetting ``NULL`` checks and would delete the
+   attribute.
+
+   .. versionchanged:: next
+      Raise an exception if called with ``NULL`` *v* and an an exception set.
```

Comment:
What exception?

Maybe reformulate it as "Must not be called with NULL as the third argument if an error was set"? SystemError is just a way to say "don't do this". Or "Always fail with NULL as the third argument if an error was set"?